controllers/argocd/route.go

```diff
@@ -302,6 +302,10 @@ func (r *ReconcileArgoCD) reconcileApplicationSetControllerWebhookRoute(cr *argo
 		}
 	}
 
+	if len(route.Spec.Host) > 63 {
+		route.Spec.Host = route.Spec.Host[:63]
+	}
+
```

Comment:
Based on what the code is doing earlier in the function, it seems like `route.Spec.Host` will always be empty at this point (correct me if I am wrong).

Reply:
Also it seems like line 328 will override this:
```
	// Allow override of the Host for the Route.
	if len(cr.Spec.Server.Host) > 0 {
		route.Spec.Host = cr.Spec.ApplicationSet.WebhookServer.Host
	}
```

Reply:
No, When I added print statement here to print `route.Spec.Host`, I am able to see the host name there(or I may be missing something, please correct me), and as it  is overriding in line number 328, I will add the logic after 328 and adding this logic `reconcileServerRoute` function as well as it is failing with same error there as well